### DISCUSSION TITLE
下载内核失败

### DISCUSSION THREAD

**Discussion by michealhansun** [UnblockNeteaseMusic/luci-app-unblockneteasemusic#112] [Q&A] [answered]
![image_2022-01-14_22-00-41](https://user-images.githubusercontent.com/9845710/149527793-a1447390-c01f-4430-b760-a77069303983.png)
![image_2022-01-14_21-59-11](https://user-images.githubusercontent.com/9845710/149527840-156d2e36-0e85-4d20-b4c2-ab7adc16e1a9.png)
![image_2022-01-14_21-59-29](https://user-images.githubusercontent.com/9845710/149527846-6090df6d-f8e0-43ec-b543-708f958f6d85.png)

**Comment by 1715173329** [comment, marked as answer]
你网烂了。

**Comment by michealhansun** [comment]
root@OpenWrt:~# unm-debug
Launching luci-app-unblockneteasmusic Debugging Tool...


OpenWrt info:
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='SNAPSHOT'
DISTRIB_TARGET='x86/64'
DISTRIB_ARCH='x86_64'
DISTRIB_TAINTS='no-all override'
DISTRIB_REVISION='R22.1.1'
DISTRIB_DESCRIPTION='OpenWrt '


uclient-fetch info:
Package: uclient-fetch
Version: 2021-05-14-6a6011df-1
Depends: libc, libuclient20201210
Provides: wget
Status: install user installed
Architecture: x86_64
Installed-Time: 1641871960

Package: libustream-openssl20201210
Version: 2020-12-10-68d09243-2
Depends: libc, libubox20210516, libopenssl1.1
Provides: libustream-openssl
Status: install user installed
Architecture: x86_64
Installed-Time: 1641871960

Downloading 'https://api.github.com/repos/UnblockNeteaseMusic/server/commits?sha=enhanced&path=precompiled'
Connecting to 192.30.255.116:443
Writing to stdout

Download completed (129380 bytes)
3cf97804c2b3e846168c5282fbddf01db52e18e5


Node.js info:
Package: node
Version: v14.18.2-1
Depends: libc, libstdcpp6, libopenssl1.1, zlib, libnghttp2-14, libatomic1
Status: install user installed
Section: lang
Architecture: x86_64
Size: 10504396
Filename: node_v14.18.2-1_x86_64.ipk
Description: Node.js® is a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node.js' package ecosystem, npm, is the largest ecosystem of open source libraries in the world. *** The following preparations must be made on the host side. *** 1. gcc 6.3 or higher is required. 2. To build a 32-bit target, gcc-multilib, g++-multilib are required. 3. Requires libatomic package. (If necessary, install the 32-bit library at the same time.) ex) sudo apt-get install gcc-multilib g++-multilib
Installed-Time: 1641871960

Node.js is placed at /usr/bin/node
Node.js version: v14.18.2


luci-app-unblockneteasmusic info:
Package: luci-app-unblockneteasemusic
Version: 2.12-2
Depends: libc, busybox, dnsmasq-full, ipset, jsonfilter, node, uclient-fetch
Status: install user installed
Architecture: all
Installed-Time: 1641871960

-rw-r--r--    1 root     root         539 Jan 14 21:58 /etc/config/unblockneteasemusic
-rwxrwxr-x    1 root     root        8.5K Jan 11 11:32 /etc/init.d/unblockneteasemusic

/usr/share/unblockneteasemusic:
drwxr-xr-x    2 root     root        4.0K Jan 14 08:31 core
-rwxrwxr-x    1 root     root        4.7K Jan 11 11:32 debugging.sh
-rwxrwxr-x    1 root     root        2.9K Jan 11 11:32 update.sh

config unblockneteasemusic 'config'
        option use_custom_cookie '0'
        option auto_update '1'
        option update_time '3'
        option local_vip '1'
        option enable_flac '1'
        option replace_music_source 'lower_than_999kbps'
        option advanced_mode '1'
        option http_port '5200'
        option https_port '5201'
        option endpoint_url 'https://music.163.com'
        option hijack_ways 'use_ipset'
        option keep_core_when_upgrade '1'
        option pub_access '1'
        option strict_mode '1'
        option netease_server_ip '59.111.19.33'
        option enable '1'
        option music_source 'default'



UnblockNeteaseMusic Node.js info:
Git HEAD version: Not Found
Core version: Not Found


Netease networking info:
< Location: http://m10.music.126.net/20220114223330/17a688e7e7fa82433ab01677f4ee9a0f/ymusic/obj/w5zDlMODwrDDiGjCn8Ky/3057940231/3d32/93a3/4db1/0757f3a378b41e09ca2767c1abd47afa.mp3
{"code":200,"data":{"music.163.com":{"ip":["45.254.48.1","101.71.154.241","59.111.19.33","59.111.160.195","193.112.159.225","47.100.127.239","39.105.175.128","39.105.63.80","182.92.170.253","118.24.63.156"],"ipv4ScoreDelay":0,"prefer":"ipv4","score":false,"ttl":300}},"msg":"OK"}

Port status:
No instance found on port 5200.
No instance found on port 5201.


Running info:
PROCD running status: inactive



Local version: NOT FOUND, latest version: 3cf97804c2b3e846168c5282fbddf01db52e18e5.
Updating core...
Failed to download app.js.

**Comment by michealhansun** [comment]
> 你网烂了。

网络问题吗？
刚刚试开了科学上网和关了科学上网都更新失败

**Comment by 1715173329** [comment]
> 刚刚试开了科学上网和关了科学上网都更新失败

这样，你先点一下删除核心，然后再更新试试

**Comment by michealhansun** [comment]
![image_2022-01-15_15-09-07](https://user-images.githubusercontent.com/9845710/149613036-454a1a88-296c-42cd-9289-d6cfa45d1745.png)
![image_2022-01-15_15-12-07-1](https://user-images.githubusercontent.com/9845710/149613133-fb4fb23e-b648-433d-bab8-79ff1e132538.png)

今日关掉IPv6，用纯IPv4 还是无法更新，请问有内核下载链接吗？ 是否可以下载后手工放进插件目录

**Comment by 1715173329** [comment]
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/app.js
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/bridge.js
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/ca.crt
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.crt
https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.key
放进 `/usr/share/unblockneteasemusic/core/`

然后 `echo '3cf97804c2b3e846168c5282fbddf01db52e18e5' > /usr/share/unblockneteasemusic/core_local_ver`

**Comment by michealhansun** [comment]
![image_2022-01-15_16-01-06](https://user-images.githubusercontent.com/9845710/149614502-e819ecf2-b391-4645-a9dc-89ad7e7799c0.png)
![image_2022-01-15_16-01-31](https://user-images.githubusercontent.com/9845710/149614504-23591d3a-7e9e-4d36-9908-81479042d41c.png)


> https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/app.js https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/bridge.js https://github.com/UnblockNeteaseMusic/server/blob/enhanced/ca.crt https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.crt https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.key 放进 `/usr/share/unblockneteasemusic/core/`
> 
> 然后 `echo '3cf97804c2b3e846168c5282fbddf01db52e18e5' > /usr/share/unblockneteasemusic/core_local_ver`

cd /usr/share/unblockneteasemusic/core/
wget https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/app.js
wget https://github.com/UnblockNeteaseMusic/server/blob/enhanced/precompiled/bridge.js
wget https://github.com/UnblockNeteaseMusic/server/blob/enhanced/ca.crt
wget https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.crt
wget https://github.com/UnblockNeteaseMusic/server/blob/enhanced/server.key

新建/usr/share/unblockneteasemusic/core_local_ver ；内容输入echo '3cf97804c2b3e846168c5282fbddf01db52e18e5'

请问是这样操作吗?还是不行，重启路由器了

**Comment by YYiiEt** [comment]
不好意思，我sb了，检查了以下发现本地dns没有劫持到clash上。。。
另外wget下载地址要换成raw格式:`https://raw.githubusercontent.com/UnblockNeteaseMusic/server/enhanced/precompiled/app.js`
不然下载会有html头导致无法识别:
![image](https://user-images.githubusercontent.com/60560292/149763251-7e380a33-f9b2-4a3d-9ddd-72a39a4a916b.png)

**Reply by michealhansun** [reply]
> 另外wget下载地址要换成raw格式:`https://raw.githubusercontent.com/UnblockNeteaseMusic/server/enhanced/precompiled/app.js`

哥们，你的问题解决了吗？我刷回之前可以正常在用的openwrt固件，也是下载内核失败，那就应该不是客户端问题，是否所有新插件都无法下载内核了？

**Reply by YYiiEt** [reply]
解决了，无法下载是我自己openclash设置的原因导致dns污染，你可以把这几个文件在win上下载再手动传上去，另外哥们/usr/share/unblockneteasemusic/core_local_ver 这个文件里面应该写入
`3cf97804c2b3e846168c5282fbddf01db52e18e5`
而不是
`echo '3cf97804c2b3e846168c5282fbddf01db52e18e5'`

**Comment by michealhansun** [comment]
哥们谢谢，可以了
